$OpenBSD: patch-kio_kio_usernotificationhandler_cpp,v 1.1 2014/06/20 13:12:36 jasper Exp $

Security fix for CVE-2014-3494, KMail/KIO POP3 SSL MITM Flaw
http://www.kde.org/info/security/advisory-20140618-1.txt

--- kio/kio/usernotificationhandler.cpp.orig	Thu Jan  2 20:26:52 2014
+++ kio/kio/usernotificationhandler.cpp	Thu Jun 19 17:00:27 2014
@@ -19,7 +19,7 @@
 #include "usernotificationhandler_p.h"

 #include "slave.h"
-#include "job_p.h"
+#include "jobuidelegate.h"

 #include <kdebug.h>

@@ -76,19 +76,18 @@ void UserNotificationHandler::processRequest()

         if (m_cachedResults.contains(key)) {
             result = *(m_cachedResults[key]);
-        } else if (r->slave->job()) {
-            SimpleJobPrivate* jobPrivate = SimpleJobPrivate::get(r->slave->job());
-            if (jobPrivate) {
-                result = jobPrivate->requestMessageBox(r->type,
-                                                      r->data.value(MSG_TEXT).toString(),
-                                                      r->data.value(MSG_CAPTION).toString(),
-                                                      r->data.value(MSG_YES_BUTTON_TEXT).toString(),
-                                                      r->data.value(MSG_NO_BUTTON_TEXT).toString(),
-                                                      r->data.value(MSG_YES_BUTTON_ICON).toString(),
-                                                      r->data.value(MSG_NO_BUTTON_ICON).toString(),
-                                                      r->data.value(MSG_DONT_ASK_AGAIN).toString(),
-                                                      r->data.value(MSG_META_DATA).toMap());
-            }
+        } else {
+            JobUiDelegate ui;
+            const JobUiDelegate::MessageBoxType type = static_cast<JobUiDelegate::MessageBoxType>(r->type);
+            result = ui.requestMessageBox(type,
+                                          r->data.value(MSG_TEXT).toString(),
+                                          r->data.value(MSG_CAPTION).toString(),
+                                          r->data.value(MSG_YES_BUTTON_TEXT).toString(),
+                                          r->data.value(MSG_NO_BUTTON_TEXT).toString(),
+                                          r->data.value(MSG_YES_BUTTON_ICON).toString(),
+                                          r->data.value(MSG_NO_BUTTON_ICON).toString(),
+                                          r->data.value(MSG_DONT_ASK_AGAIN).toString(),
+                                          r->data.value(MSG_META_DATA).toMap());
             m_cachedResults.insert(key, new int(result));
         }
     } else {
